### PR TITLE
DAPT codegen data curation tutorial

### PR DESCRIPTION
**Pull request by jvamaraju**
## Description

Added a tutorial to curate data for DAPT codegen use-case with Nemo Curator

## Usage

```python
codegen_DAPT\llm_data_collection\notebooks\data_curation-nemo-curator_DAPT.ipynb
```
## Checklist

- I am familiar with the [Contributing Guide](https://github.com/NVIDIA/NeMo-Curator/blob/main/CONTRIBUTING.md).
- New or Existing tests cover these changes.
- The documentation is up to date with these changes.

### CODE REVIEW

**Comment by Maghoumi** [comment]
@jvamaraju Thank you for your contribution. Here's some initial feedback after spending some time going through the PR.

### High level comments

Considering that this is a tutorial, we need to have explicit entry points for each topic where we are attempting to teach users how to accomplish certain things, or show them the best practices. We can follow a structure roughly similar to what you walked us through in the weekly meetings. Consider these (hypothetical) steps. I think the tutorial greatly benefits from having a "flow" where users can experience each data curation step by going through the flow:

* **Step 1: Dependency Installation**
  - Before users do anything, they need to understand all the code dependencies they need to fulfill. This includes pip package installs, PDF cleaner installation, etc. We could either have a notebook for this or a README file. I think a notebook would be better due to its interactive nature.

* **Step 2: Data Acquisition**
  - Substep 1: Users need to download the data from the Google Drive share. Show them where the data is located, and what directory this needs to be under.
  - Substep 2: Users need to run some shell scripts to traverse the downloaded data and create the relevant meta data.
  - ... [The rest follows] ...

* **Step 3: Data Curation using NeMo Curator**
  - Substep 1: Install dependencies
  - Substep 2: .....

* **Step 4: Data Curation without NeMo Curator**
  - Substep 1: .....
  - Substep 2: .....

Specifically, I think the data acquisition step also deserves a dedicated notebook where users can walk through the sequence of operations they need to do before they can follow along with the rest of the tutorial. Also, if there are shell scripts that need execution, we can make cells for them inside the notebook using `%` operator ([see this example](https://stackoverflow.com/a/70398447/398316)). This way, users won't have to worry about the order of executing different scripts.

Lastly, where relevant, I suggest clearing the cell outputs from the notebooks. I found myself scrolling a lot while going through the code and understanding the structure. For instance, [this notebook](https://github.com/jvamaraju/ndc_dapt_playbook/blob/main/tutorials/dapt-curation/codegen_DAPT/llm_data_collection/notebooks/data_curation-nemo-curator_DAPT.ipynb), is around 2MB and most cell outputs contain logs that users would be able to reproduce if they ran it locally. We should be safe to remove all those outputs from the checked in file. In some cases, it might make sense to keep that output (e.g. when we're trying to show a sample output to the user), so it's fine to keep them.

### Organization
The tutorial consists of 4 main types of files:
1. Python scripts
2. Shell scripts
3. Notebook files
4. Data files consisting of
   1. Images (e.g. PNG)
   2. JSONL (e.g. [this one](https://github.com/jvamaraju/ndc_dapt_playbook/blob/main/tutorials/dapt-curation/codegen_DAPT/llm_data_collection/notebooks/github_repos_opensource.jsonl))
   3. Data that the user downloads.

One challenge I faced while going through the repo was that different files appear in different places. In order to greatly enhance the readability and organization, I suggest the following structure (this is a rough sketch, feel free to incorporate based on your intuition of course!) :
* Aggregate all scripts inside one folder (e.g. `scripts`). It can have subfolders to organize based on the step of data curation (e.g. `scripts/1-data-acquisition` or `scripts/2-data-curation`).
* Organize Python scripts into logical units. For instance, if there are scripts that are helpers, they can all go inside a `helpers` directory. Reusable code can go into a `common` folder for instance.
* Aggregate all images in the same place. This has been already done, but some PNG files are still floating around.
* Create a centralized place for all other types of data. Perhaps the `dataset` folder that you have can serve as this? I'm thinking we could create a specific folder inside it `dataset/downloaded` where users can download the dataset under. Then everything else can reside in `dataset` directly (e.g. JSONL files).

### PDF Cleaner
This is a very useful tool and is standalone. I think it can be reused over and over throughout the repo. In its current form, it looks like it is a step of the tutorial. I'll defer to @ayushdg or others to comment on the best organization for this. It also contains docker image creation, etc. which appear to be out of the scope of the tutorial. Let's brainstorm on the most logical way of integrating this tool into the repo.

**Comment by jvamaraju** [comment]
@Maghoumi  I have incorporated all the feedback. Specifically
Added license headers to all the files.
Cleaned up and unify function names
Added docstrings to all functions
Thoroughly tested with all the sources (currently it limits to 10 sources of Wikipedia, GitHub and arXiv).

Please review.

File: tutorials/dapt-curation/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+* `\code\data` contains curated data after processing
+
+* Notebook for data curation with NeMo Curator: `\code\DAPT_data_curation.ipynb`
+
+The small size of this dataset makes it ideal for creating and validating data curation pipelines on a local machine or a cluster.
+
+This playbook utilizes specific tools and techniques. First, we convert all files to Txt format (if not already in Txt), compress files on disk, add metadata, and convert them to JSON (`./data/raw/`). Then, we leverage [NeMo Curator](https://github.com/NVIDIA/NeMo-Curator/tree/main) to mine high-quality text at scale from a massive code-generation corpus. [NeMo Curator](https://github.com/NVIDIA/NeMo-Curator/tree/main), built on Dask and RAPIDS, is instrumental in scaling data curation and providing GPU acceleration. We use its capabilities to extract text, identify code file types, fix unicode errors, filter quality through heuristics, deduplicate, and redact personal information. We finally also provide steps to blend and shuffle data sources for continued pre-training.
+
+
+## NeMo Tools and Resources
```
Comment (Maghoumi) [inline tutorials/dapt-curation/README.md:18]: Perhaps this is not needed? Because it's already inside the GitHub repo.

File: tutorials/dapt-curation/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+* [NeMo-Curator GitHub repo](https://github.com/NVIDIA/NeMo-Curator/tree/main)
+
+
+## Software Requirements
```
Comment (Maghoumi) [inline tutorials/dapt-curation/README.md:23]: The installation instructions are on the main README file already, so I suggest removing this as well to prevent the need for keeping it updated.

File: tutorials/dapt-curation/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+* This playbook has been tested on: nvcr.io/nvidia/nemo:24.03.01.framework. It is expected to work similarly on other environments. Within the container, from the NeMo-Curator directory install using `pip install --extra-index-url https://pypi.nvidia.com ".[cuda12x]"`
```
Comment (Maghoumi) [inline tutorials/dapt-curation/README.md:24]: Let's add a note that this can be executed on a local machine with a GPU, but it has been tested with .....
Same as above, let's remove the install instructions.

File: tutorials/dapt-curation/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+* NeMo Curator currently requires Python 3.10 and the GPU accelerated modules require CUDA 12 or above installed in order to be used.
+
+
+## Hardware Requirements
+* This playbook can run on CPUs or GPUs. For GPUs, this playbook has been tested on minimum 1xA100 80G
+
+
+## Walkthrough
+For a detailed walkthrough of this tutorial, please see this blog post.
```
Comment (Maghoumi) [inline tutorials/dapt-curation/README.md:33]: Let's remove this for now. I can add this back once the blog is published.

File: tutorials/distributed_data_classification/distributed_data_classification.ipynb
Comment (Maghoumi) [inline tutorials/distributed_data_classification/distributed_data_classification.ipynb:1]: [For this and other files below that are irrelevant to the tutorial] I think the permission values changed? Could you please undo the changes to all these files and push again? Ideally, this PR shouldn't modify any irrelevant parts of the repo.

File: tutorials/dapt-curation/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+[ChipNeMo](https://arxiv.org/pdf/2311.00176) is a chip design domain adapted LLM. LLama models are continually pre-trained with 20B plus tokens on domain-specific chip design data, including code, documents, etc., based on NeMo foundation models and then fine-tuned with instruction datasets from design data as well as external sources.
+
+Here, we share a tutotial with best practices on data curation for DAPT (domain-adaptive pre-training) for a ChipNeMo-like code generation use case.
+
+* In this tutorial, we will leverage chip domain/hardware datasets from open-source GitHub repositories (`.code/sources/github_repos.jsonl`), wiki URLs (`.code/sources/wikipedia_urls.jsonl`), and academic papers(`./sources/arxiv_urls.jsonl`).
+
+* `\code\data` contains curated data after processing
+
+* Notebook for data curation with NeMo Curator: `\code\DAPT_data_curation.ipynb`
```
Comment (Maghoumi) [inline tutorials/dapt-curation/README.md:11]: I'm a little confused about this: IIRC, the notebook version had issues, so we decided to publish this in the script format (`main.py`). If my understanding is correct, what's the reason for keeping the notebook?

If this is for content only, we can move the content to the blog.

File: tutorials/dapt-curation/README.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+The small size of this dataset makes it ideal for creating and validating data curation pipelines on a local machine or a cluster.
```
Comment (Maghoumi) [inline tutorials/dapt-curation/README.md:13]: Is the dataset really small?

File: tutorials/dapt-curation/code/sources/github_repos.jsonl [hunk 7]
```diff
@@ -0,0 +1,6 @@
+"google/skywater-pdk-libs-sky130_fd_sc_lp"
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/sources/github_repos.jsonl:1]: Didn't we have way more repositories in this file? Did we remove for simplicity, or was there some other reason?

Asking because if someone wants to repro ChipNeMo data exactly, they'd need all the repos that was used for model training.

File: tutorials/dapt-curation/code/utils.py [hunk 8]
```diff
@@ -0,0 +1,389 @@
...
+                RepeatingTopNGramsFilter(n=3, max_repeating_ngram_ratio=0.18),
+                text_field="text",
+                score_type=float,
+            ),
+            ScoreFilter(
+                RepeatingTopNGramsFilter(n=4, max_repeating_ngram_ratio=0.16),
+                text_field="text",
+                score_type=float,
+            ),
+            #             # If the document shrinks by > 30% in terms of number of lines after
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:131]: Either uncomment or remove entirely.

File: tutorials/dapt-curation/code/utils.py [hunk 9]
```diff
@@ -0,0 +1,389 @@
...
+            # Unify all the quotation marks
+            Modify(QuotationUnifier(), text_field="text"),
+            # Unify all unicode
+            Modify(UnicodeReformatter(), text_field="text"),
+        ]
+    )
+    return cleaners(dataset)
+
+
+def filter_txt(dataset: DocumentDataset) -> DocumentDataset:
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:91]: [nit] rename to `filter_text_sources` so that it's more descriptive?

File: tutorials/dapt-curation/code/utils.py [hunk 10]
```diff
@@ -0,0 +1,389 @@
...
+
+    """
+    data = json.loads(json_data)
+    json_docs = data.get("documents", [])
+    print("len of docs: ", len(json_docs))
+    print(json_docs)
+    for doc in json_docs:
+        print(doc)
+        if "text" in doc:
+            # print("initial", doc["text"])
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:236]: Remove

File: tutorials/dapt-curation/code/utils.py [hunk 11]
```diff
@@ -0,0 +1,389 @@
...
+
+                # Get the first (and presumably only) value from the Series
+                text_value = text_series.iloc[0]
+
+                # Convert the value to a string
+                redacted_text_string = str(text_value)
+
+                doc["text"] = doc["text"].replace(comment, redacted_text_string)
+
+                # print("final", doc["text"])
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:266]: Remove

File: tutorials/dapt-curation/code/utils.py [hunk 12]
```diff
@@ -0,0 +1,389 @@
...
+                text_field="text",
+                score_type=int,
+            ),
+        ]
+    )
+    filtered_dataset = filters(dataset)
+    return filtered_dataset
+
+
+# +
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:223]: Remove `#`

File: tutorials/dapt-curation/code/utils.py [hunk 13]
```diff
@@ -0,0 +1,389 @@
...
+            end_idx = doc["text"].find("*/")
+            if start_idx != -1 and end_idx != -1:
+                comment = doc["text"][start_idx : end_idx + 2]
+
+                # Create a Dask DataFrame from the given data
+                comment_df = dd.from_pandas(
+                    pd.DataFrame([{"text": comment}]), npartitions=1
+                )
+                # Create a DocumentDataset instance
+                comment_doc_dataset = DocumentDataset(comment_df)
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:249]: I'm a little confused as why this is being done this way. Couldn't we do this as part of the `run_curation_pipeline`? My concern is that this will have poor performance.

File: tutorials/dapt-curation/code/utils.py [hunk 14]
```diff
@@ -0,0 +1,389 @@
...
+    Returns:
+        DocumentDataset: The redacted dataset with PII replaced by a generic value.
+    """
+    DEFAULT_MAX_DOC_SIZE = 3000000
+    redactor = Modify(
+        PiiModifier(
+            supported_entities=[
+                "PERSON",
+                "EMAIL_ADDRESS",
+                #                                "URL",
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:322]: Either uncomment, or remove entirely.

File: tutorials/dapt-curation/code/utils.py [hunk 15]
```diff
@@ -0,0 +1,389 @@
...
+                #                                "PHONE_NUMBER",
+                #                                "ADDRESS",
+                #                                "LOCATION"
+            ],
+            anonymize_action="replace",
+            #             batch_size=1000,
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:328]: Remove?

File: tutorials/dapt-curation/code/utils.py [hunk 16]
```diff
@@ -0,0 +1,389 @@
...
+    return DocumentDataset(deduped)
+
+
+# +
+def filter_txt_lines(dataset: DocumentDataset) -> DocumentDataset:
+    """
+    Discard text files based on lines.
+    """
+    dataset_df = dataset.df
+    #     print(len(dataset_df.index))
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/utils.py:368]: Remove?
Since there appears to be multiple occurances, I won't comment further on this, please remove all debugging code from the PR for cleaning purposes.

File: tutorials/dapt-curation/code/main.py [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+    get_all_files_paths_under,
+    separate_by_metadata,
+)
+from nemo_curator.utils.script_utils import add_distributed_args, parse_client_args
+
+SCRIPT_DIR_PATH = os.path.dirname(os.path.abspath(__file__))
+DATA_DIR = os.path.join(SCRIPT_DIR_PATH, "data")
+
+#
+# TODO
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/main.py:58]: Could you add a note about this in the README file?

File: tutorials/dapt-curation/code/main.py [hunk 18]
```diff
@@ -0,0 +1,245 @@
...
+    )
+
+    # Define data curation steps for code files
+    curation_steps_code = Sequential(
+        [
+            dedupe,
+            filter_code_lines,
+            filter_code,
+            clean_and_unify,
+            #          redact_pii,
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/main.py:121]: Remove?

File: tutorials/dapt-curation/code/main.py [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+    )
+
+    orig_dataset_text = DocumentDataset.read_json(text_files, add_filename=True)
+    orig_dataset_code = DocumentDataset.read_json(code_files, add_filename=True)
+
+    # visualize file types and sizes
+    orig_df = orig_dataset_text.df.compute()
+    orig_df = orig_df.reset_index()
+
+    # Create a histogram for different file types -text
```
Comment (Maghoumi) [inline tutorials/dapt-curation/code/main.py:132]: Let's move the logic for showing histograms, etc to its own function so that the code looks cleaner.

Something like `def plot_data(...)`

**Review by Maghoumi — COMMENTED** [review]
Thanks again for your contribution. I took one pass and left some comments.

File: tutorials/dapt-curation/README.md (hunk 5 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/README.md:11]: @Maghoumi The notebook is just for content and quick tests. It will also help with any demos as we want to walk through the code. Should we still remove it?

File: tutorials/dapt-curation/README.md (hunk 6 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/README.md:13]: Yeah its extremely tiny compared to full dapt. We use only 100 wiki links, 10 papers and 6 repos. Chipnemo was trained on much more data.

File: tutorials/dapt-curation/code/sources/github_repos.jsonl (hunk 7 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/sources/github_repos.jsonl:1]: @Maghoumi We cannot share the full data that ChipNeMo used because of legal permissions. So, we had to sample only a few datasets where permissions to share was easy to get.

File: tutorials/dapt-curation/code/utils.py (hunk 9 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/utils.py:91]: We have filter_code and filter_txt. I think that's descriptive enough?

File: tutorials/dapt-curation/code/utils.py (hunk 13 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/utils.py:249]: It has poor performance. But also, redaction has a few bugs when used for code files. We discussed this with Arham as well. The idea is to redact only the comment lines and not all the code. We saw that proper names within code get redaxted/replaced otherwise. Is there an alternate way?

File: tutorials/dapt-curation/code/utils.py (hunk 14 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/utils.py:322]: Removed it, but it shows the developer the other possible options as well

File: tutorials/dapt-curation/code/utils.py (hunk 15 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/utils.py:328]: This is a good toggle option to keep. Again, points to hyperparameters users need to play with.

File: tutorials/dapt-curation/code/utils.py (hunk 16 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/utils.py:368]: done

File: tutorials/dapt-curation/code/main.py (hunk 18 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/main.py:121]: Lets keep this for now. I'm testing it even more for performance.

File: tutorials/dapt-curation/code/main.py (hunk 19 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/main.py:132]: done

File: tutorials/dapt-curation/code/main.py (hunk 17 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/main.py:58]: done

**Comment by jvamaraju** [comment]
@Maghoumi  Please look at my comments and updated repo.

File: tutorials/distributed_data_classification/distributed_data_classification.ipynb
Comment (jvamaraju) [reply tutorials/distributed_data_classification/distributed_data_classification.ipynb:1]: Can you check now? I did not change permissions I thought. Just synced the repo with main one again. Let me know if you still see this issue

File: tutorials/dapt-curation/README.md (hunk 5 above)
Comment (Maghoumi) [reply tutorials/dapt-curation/README.md:11]: Yes. I suggest keeping a copy of it in a branch or something for testing.
Given that this repo is production-grade, we're not allowed to push experimental/debugging content to it.

File: tutorials/dapt-curation/README.md (hunk 6 above)
Comment (Maghoumi) [reply tutorials/dapt-curation/README.md:13]: Since the goal was to reproduce ChipNeMo's data curation, shouldn't we still include all the sources?
We can use the `limit=10` for the downloaders (along with a note for users) to maintain the ability of running the whole thing on a local machine.

File: tutorials/dapt-curation/code/main.py (hunk 17 above)
Comment (Maghoumi) [reply tutorials/dapt-curation/code/main.py:58]: Did you also remove the TODO notes here? If already removed, it's not showing up here for some reason.

File: tutorials/dapt-curation/code/sources/github_repos.jsonl (hunk 7 above)
Comment (Maghoumi) [reply tutorials/dapt-curation/code/sources/github_repos.jsonl:1]: I thought the legal issue was with bundling all the data into the Google Drive share. Are you saying we can't include a link to the repositories, Wikipedia pages and arXiv papers?

File: tutorials/dapt-curation/code/utils.py (hunk 9 above)
Comment (Maghoumi) [reply tutorials/dapt-curation/code/utils.py:91]: Yes, but let's rename to `filter_text`.

File: tutorials/dapt-curation/code/utils.py (hunk 13 above)
Comment (Maghoumi) [reply tutorials/dapt-curation/code/utils.py:238]: Adding @ryantwolf and @ayushdg to comment on how to workaround this issue.

File: tutorials/dapt-curation/code/utils.py (hunk 15 above)
Comment (Maghoumi) [reply tutorials/dapt-curation/code/utils.py:311]: Then we should add `if .... else.... ` block. Leaving commented out code is not allowed in production grade code.

File: tutorials/distributed_data_classification/distributed_data_classification.ipynb
Comment (Maghoumi) [reply tutorials/distributed_data_classification/distributed_data_classification.ipynb:1]: These files are still showing as changed. Here's how I would resolve it:

```
git branch backup  # Make a backup of your branch as-is in case something gets messed up.
git reset 224800b4f141b41cdea376daf3c55a0cd788df47~1  # Undo your entire commit history, but keeps changed files.

git add tutorials/dapt-curation
git commit -sS -m "Your commit message"
git reset --hard  # to get rid of irrelevant changes
git push -f
```

Basically the idea is to only add the stuff under `tutorials/dapt-curation` to your branch, then commit everything as a new commit. Let me know if this makes sense.

File: tutorials/dapt-curation/code/utils.py (hunk 14 above)
Comment (Maghoumi) [reply tutorials/dapt-curation/code/utils.py]: Thanks.
The usage is already documented and shown in other tutorials.

File: tutorials/dapt-curation/code/utils.py (hunk 13 above)
Comment (ryantwolf) [reply tutorials/dapt-curation/code/utils.py:238]: As far as I know the PII tool was made to operate on normal text documents and not on code. In particular, it assumes it can operate on the entire document. So, there will probably need to be a workaround. However, I can suggest on how to improve performance. I would recommend you construct the entire `DocumentDataset` filled with every comment in your dataset. Then, perform the PII on the comments, then replace them. Doing this in a batched manner should allow the PII removal to perform better. @singhva if you have any additional thoughts

**Comment by jvamaraju** [comment]
[like]  Janaki Vamaraju reacted to your message:
________________________________

File: tutorials/distributed_data_classification/distributed_data_classification.ipynb
Comment (jvamaraju) [reply tutorials/distributed_data_classification/distributed_data_classification.ipynb:1]: Ok I did this. It seemed to have worked. Please let me know if you see it.

File: tutorials/dapt-curation/README.md (hunk 5 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/README.md:11]: Ok I added it through .gitignore. Will this work?

File: tutorials/dapt-curation/README.md (hunk 6 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/README.md:13]: The issue is we are legally not allowed to share all the sources in any format. So we are limited to this. This replicates data curation on a small, diversified dataset and is representative.

File: tutorials/dapt-curation/code/sources/github_repos.jsonl (hunk 7 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/sources/github_repos.jsonl:1]: Yes exactly. Not all github repos and arxiv papers have permissions for us to share. So I had to change to a few sources where we have permissions.

File: tutorials/dapt-curation/code/utils.py (hunk 9 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/utils.py:90]: done

File: tutorials/dapt-curation/code/utils.py (hunk 15 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/utils.py:310]: ok removed it. Its a tutorial right? But i understand.

File: tutorials/dapt-curation/code/utils.py (hunk 13 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/utils.py]: Thanks @ryantwolf  and @Maghoumi . I have updated the PII reaction piece now.

File: tutorials/dapt-curation/code/main.py (hunk 18 above)
Comment (jvamaraju) [reply tutorials/dapt-curation/code/main.py]: Added it in.

**Comment by jvamaraju** [comment]
pre-commit.ci autofix

**Comment by jvamaraju** [comment]
pre-commit.ci autofix

**Comment by ryantwolf** [comment]
Closing in favor of #160
